packages/retail-ui/components/DropdownContainer/DropdownContainer.tsx

```diff
@@ -182,22 +168,56 @@ export default class DropdownContainer extends React.Component<
         top,
         left,
         right,
-        bottom,
-        minWidth: targetRect.right - targetRect.left
+        bottom
       };
-      this.setState({ position });
+
+      this.setState({
+        position: this.props.disablePortal
+          ? this.convertToRelativePosition(position)
+          : position
+      });
     }
   };
 
-  private _getHeight = () => {
-    if (!this._dom) {
+  private getHeight = () => {
+    if (!this.dom) {
       return 0;
     }
-    const child = this._dom.children.item(0);
+    const child = this.dom.children.item(0);
     if (!child) {
       return 0;
     }
     const rect = child.getBoundingClientRect();
     return rect.height ? rect.height : rect.bottom - rect.top;
   };
+
+  private getMinWidth = () => {
+    const target = this.props.getParent() as Nullable<Element>;
```

Comment:
Старайся не использовать `as`. `as` не кастует к нужному тебе типу, так ты просто пытаешься обмануть TypeScript, говоря ему, что ты уверен в том что здесь всегда будет значение такого типа. В таких ситуациях возможно будет лучше изменить тип у getParent?

Reply:
Тип тоже не просто изменить. Проблема перейдет в те компоненты, что используют `DropdownContainer`.

Предлагаю:
 ```suggestion
  private isElement = (node: Element | Text | null): node is Element => {
    return node instanceof Element;
  };
```

Reply:
Да, тоже хороший вариант

Reply:
fixed